service/src/main/kotlin/fi/espoo/evaka/attendance/RealtimeStaffAttendanceController.kt

```diff
@@ -180,8 +180,12 @@ class RealtimeStaffAttendanceController(private val accessControl: AccessControl
             staffAttendances.all { it.groupId != null || !it.type.presentInGroup() }
 
         fun anyAttendanceAfter(timestamp: HelsinkiDateTime): Boolean =
-            staffAttendances.any { it.arrived > timestamp } &&
-                externalAttendances.any { it.arrived > timestamp }
+            staffAttendances.any {
+                it.arrived > timestamp || (it.departed != null && it.departed > timestamp)
+            } ||
+                externalAttendances.any {
+                    it.arrived > timestamp || (it.departed != null && it.departed > timestamp)
+                }
```

Comment:
I thought it would be very nice to use range operations for these checks, so I added `overlaps()` for `TimeInterval` in #4827.

These checks would then become just `TimeInterval(it.arrived, it.departed).overlaps(TimeInterval(timestamp, null))` and the delicate range logic doesn't have to be duplicated.

Reply:
Actually these are HelsinkiDateTimes, so TimeInterval does not work